pandas/core/groupby/groupby.py

```diff
@@ -508,13 +508,16 @@ def _get_index(self, name):
     @cache_readonly
     def _selected_obj(self):
         # Note: _selected_obj is always just `self.obj` for SeriesGroupBy
-
-        if self._selection is None or isinstance(self.obj, Series):
+        if hasattr(self, "mutated") and self.mutated:
```

Comment:
`if getattr(self, 'mutated', None/False):`

Reply:
Thanks for the tip @mroeschke , that's much nicer!